questdb-rs/src/ingress/mod.rs

```diff
@@ -409,44 +409,51 @@ fn write_escaped_impl<Q, C>(
     output: &mut String,
     s: &str)
         where
-            C: Fn(char) -> bool,
-            Q: Fn(&mut String) -> ()
+            C: Fn(u8) -> bool,
+            Q: Fn(&mut Vec<u8>) -> ()
 {
+    let output_vec = unsafe { output.as_mut_vec() };
     let mut to_escape = 0usize;
-    for c in s.chars() {
-        if check_escape_fn(c) {
+    for b in s.bytes() {
+        if check_escape_fn(b) {
             to_escape += 1;
         }
     }
 
-    quoting_fn(output);
+    quoting_fn(output_vec);
 
     if to_escape == 0 {
-        output.push_str(s);
+        // output.push_str(s);
+        output_vec.extend_from_slice(s.as_bytes());
     }
     else {
-        output.reserve(s.len() + to_escape);
-        for c in s.chars() {
-            if check_escape_fn(c) {
-                output.push('\\');
+        let additional = s.len() + to_escape;
+        output_vec.reserve(additional);
+        let mut index = output_vec.len();
+        unsafe { output_vec.set_len(index + additional) };
+        for b in s.bytes() {
+            if check_escape_fn(b) {
+                output_vec[index] = b'\\';
```

Comment:
You can go even further: `unsafe { *output_vec.get_unchecked_mut(index) = b'\\'; }`

Reply:
Ah! I forgot the index was checked! Thanks!

Reply:
That was quite a perf boost too. Now the function is almost 30% faster compared to the pre-PR implementation.